pkg/sources/gitlab/gitlab.go

```diff
@@ -203,6 +213,68 @@ func (s *Source) Chunks(ctx context.Context, chunksChan chan *sources.Chunk, _ .
 	return s.scanRepos(ctx, chunksChan)
 }
 
+func (s *Source) scanTargets(ctx context.Context, targets []sources.ChunkingTarget, chunksChan chan *sources.Chunk) error {
+	for _, tgt := range targets {
+		if err := s.scanTarget(ctx, tgt, chunksChan); err != nil {
+			ctx.Logger().Error(err, "error scanning target")
+		}
+	}
+
+	return nil
+}
+
+func (s *Source) scanTarget(ctx context.Context, target sources.ChunkingTarget, chunksChan chan *sources.Chunk) error {
+	metaType, ok := target.QueryCriteria.GetData().(*source_metadatapb.MetaData_Gitlab)
+	if !ok {
+		return fmt.Errorf("unable to cast metadata type for targeted scan")
+	}
+	meta := metaType.Gitlab
+	projID, sha := int(meta.GetProjectId()), meta.GetCommit()
+	if projID == 0 || sha == "" {
+		return fmt.Errorf("project ID and commit SHA must be provided for targeted scan")
+	}
+
+	logger := ctx.Logger().WithValues("project_id", projID, "commit", sha)
```

Comment:
❤️ this. i have a general question about the way we use this pattern though: why do we create only loggers when we do this, rather than entirely new contexts? to me, making a new context seems preferable because it's easier to propagate these logging kvps to called functions. (you didn't do that here - was it on purpose?) but i see it done so often the other way that i suspect i'm missing something.

Reply:
Not sure, I just noticed we did it elsewhere so I did the same. Updated to pass a ctx instead.